src/main/java/org/jenkinsci/plugins/GithubSecurityRealm.java

```diff
@@ -400,6 +411,23 @@ public HttpResponse doFinishLogin(StaplerRequest request)
         return HttpResponses.redirectToContextRoot();   // referer should be always there, but be defensive
     }
 
+    private void storeCredentials(String accessToken, GithubAuthenticationToken auth) throws IOException {
```

Comment:
For credentials, this should be using `hudson.util.Secret` instead of `String` if I understand this correctly.  That way, when the configuration is serialized to XML on disk it, the value will be encrypted.  `Secret` behaves like a normal `String` but has the added benefits of encrypting the value before writing to disk.